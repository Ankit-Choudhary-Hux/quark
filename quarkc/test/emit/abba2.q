quark *;
 // xfail circular import in py, works in java, js, rb
namespace abba2 {
    namespace c {
        class C {
            void foo() { print(self.getClass().getName()); }
        }
    }
    namespace a {
        class A extends c.C {}
    }
    namespace b {
        class B extends c.C {}
        class BA extends a.A {}
    }
}
void main(List<String> args) {
    new abba2.b.BA().foo();
}
