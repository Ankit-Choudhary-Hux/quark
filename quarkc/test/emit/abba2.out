abba2.b.BA
